src/controllers/profile.controller.js

```diff
@@ -0,0 +1,18 @@
+const { ApiError } = require('../exceptions/api.error');
+
+const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
```

Comment:
There is a logical flaw here. The code should include an 'else' statement before the default error response to prevent sending multiple responses. Without the 'else', the code will attempt to send two responses for ApiError instances, which will cause an error.